jax/_src/numpy/linalg.py

```diff
@@ -531,10 +548,14 @@ def norm(x: ArrayLike, ord: Union[int, str, None] = None,
     raise ValueError(
         f"Invalid axis values ({axis}) for jnp.linalg.norm.")
 
+@overload
+def qr(a: ArrayLike, mode: Literal["r"]) -> Array: ...
+@overload
+def qr(a: ArrayLike, mode: str = "reduced") -> Union[Array, Tuple[Array, Array]]: ...
```

Comment:
It maybe could (actually it would be `Union[Literal["reduced'], Literal["complete"], Literal["r"], Literal["r, but then if someone tries to pass a string variable to the function it would fail unless it's explicitly annotated as a `Literal` rather than a string, which is kind of inconvenient

Reply:
I see, thanks!